SwiftGit2/Repository.swift

```diff
@@ -509,4 +509,23 @@ final public class Repository {
 	                     progress: CheckoutProgressBlock? = nil) -> Result<(), NSError> {
 		return setHEAD(reference).flatMap { self.checkout(strategy: strategy, progress: progress) }
 	}
+	
+	/// Cache for commits(in: branch)
+	///
+	/// Specifically for allowing syntax "while let commit = repo.commits(in: branch).next() {}"
+	public var branchCommitIteratorMap: [Branch: CommitIterator] = [:]
```

Comment:
This has to go, unfortunately. Multiple readers should be able to read the same branch simultaneously. This would seriously complicate that.

But… This should do what you want:

```swift
for commit in repo.commits(in: branch) {}
```